cli/src/semgrep/commands/lsp.py

```diff
@@ -23,7 +23,7 @@ def init_log(name: str, level: int, logfile: Optional[str] = None) -> None:
     log.addHandler(handler)
 
 
-@click.command()
+@click.command(hidden=True)
```

Comment:
there is no option to hide the visibility of flags in osemgrep, so this modification is useless.
The help text for command is automatically generated by the Cmdliner OCaml library which generates a man page
.

Reply:
This flag controls the visibility of the subcommand when running `pysemgrep`'s help not the command's help text.

Reply:
We don't want to hide lsp as a subcommand. cc @ajbt200128

Reply:
this was the direction from design

<img width="311" alt="Screenshot 2023-08-25 at 2 06 51 PM" src="https://github.com/returntocorp/semgrep/assets/5779832/11298401-d483-420d-9ae4-9b899d6d22df">

cc @adamboros

Reply:
Yea, I'm not sure about the publish flag but I don't see why we'd want to hide the lsp command. If anything we should display it AND link to the editor integrations page

Reply:
@aryx can you details why we would not want to hide the lsp subcommand?

Reply:
I don't know when this design decision was discussed, but I think it would be better next time to involve me or austin about this kind of discussion as we have more context about how semgrep CLI is used.

Reply:
To answer your question bruno, it's just what austin said, we actually want to advertise semgrep lsp more, so people know about it.